src/coreclr/nativeaot/BuildIntegration/BuildFrameworkNativeObjects.proj

```diff
@@ -8,6 +8,9 @@
     <BuildingFrameworkLibrary>true</BuildingFrameworkLibrary>
     <Optimize Condition="'$(Configuration)' == 'Release' and '$(Optimize)' == ''">true</Optimize>
     <DebugSymbols Condition="'$(DebugSymbols)' == ''">true</DebugSymbols>
+    <_targetOS>$(TargetOS)</_targetOS>
```

Comment:
I assume that it is not easy to fix the tests scripts to pass it RuntimeIdentifier. Correct?

(It is ok to take this shortcut since this is only used by the test scripts.)

Reply:
Actually the tests are passing `RuntimeIdentifier` here. The issue is that tests set `RuntimeIdentifier=OutputRid`. The OutputRid in case of non-portable build is distro-specific. I think `_targetArchitecture` is needed here and `_targetOS` is optional. I will give it a try.

Reply:
Without `_targetOS` setting here, `windows-x64 Checked NativeAOT` failed because it directly calls `Microsoft.NETCore.Native.targets` (without going through `Microsoft.DotNet.ILCompiler.SingleEntry.targets`) and `_targetOS` is empty which then includes `Microsoft.NETCore.Native.Unix.targets` instead of `Microsoft.NETCore.Windoes.targets`.